src/main/java/org/takes/facets/forward/TkForward.java

```diff
@@ -107,21 +111,23 @@ public InputStream body() throws IOException {
          * @throws IOException If fails
          */
         private Response load() throws IOException {
-            synchronized (this.saved) {
-                if (this.saved.isEmpty()) {
-                    Iterable<String> head;
-                    InputStream body;
-                    try {
-                        head = this.origin.head();
-                        body = this.origin.body();
-                    } catch (final RsForward ex) {
-                        head = ex.head();
-                        body = ex.body();
+            if (this.saved.get() == null) {
+                synchronized (this.mutex) {
+                    if (this.saved.get() == null) {
```

Comment:
@yegor256 because `CopyOnWriteArrayList` is already thread safe so synchronizing access to a thread safe object is considered as a violation by FindBugs as it sounds like a misuse but if you prefer that I put it back with and Ignore I can, even if I personally find the previous approach much worse than the one I proposed, indeed using a `CopyOnWriteArrayList` to add only one element and return the first element sounds really like an ugly hack to me, it is only my point of view. The only advantage of the old approach compared to mine is the fact that you don't pay the price of an access to a `volatile` variable so it will be slightly faster but that's all. So what should I do? Revert the changes and ignore it?